app/uk/gov/hmrc/tariffclassificationfrontend/views/cancel_ruling.scala.html

```diff
@@ -47,11 +45,7 @@ <h3 class="heading-large mt-0">Cancel the ruling</h3>
                 <li><a href="@uk.gov.hmrc.tariffclassificationfrontend.controllers.routes.AttachmentsController.attachmentsDetails(c.reference)">attach</a>  any relevant documentation to the case</li>
             </ul>
 
-            @form(action = uk.gov.hmrc.tariffclassificationfrontend.controllers.routes.CancelRulingController.confirmCancelRuling(c.reference), 'class -> "form") {
-                @CSRF.formField
-                <button class="button" id="cancel-ruling-button" type="submit">Cancel ruling</button>
-                <a class="secondary-button" id="cancel-button" href="@uk.gov.hmrc.tariffclassificationfrontend.controllers.routes.CaseController.rulingDetails(c.reference)">Cancel</a>
-            }
+            @cancel_ruling_form(c, form("reason"))
```

Comment:
I think this cancel_ruling exists because the form to cancel then separate it into another class (is not bad), how granular we want to be?? My opinion is always the clearest and easy to maintain option. But I think that having a form object and parsing just a field of this form into another component is a good reason to not have a separate component. What do you think?

Reply:
No strong opinion either way.  As the cancel form component is not really re-usable its contents could have been in-lined.   My defence is that I was following patterns seen elsewhere.

Reply:
I dont see any harm pulling it out. although I agree its not particularly reusable